### PR TITLE
Fix email address validation

### PR DESCRIPTION
**Pull request by gbirke**
Don't validate email address data for anonymous donors when adding
donations.

### CODE REVIEW

File: tests/Fixtures/NullDomainNameValidator.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\DonationContext\Tests\Fixtures;
+
+use WMDE\FunValidators\DomainNameValidator;
+
+/**
+ * @license GNU GPL v2+
+ */
+class NullDomainNameValidator implements DomainNameValidator {
```
Comment (JeroenDeDauw) [inline tests/Fixtures/NullDomainNameValidator.php]: I'd go for SucceedingDomainNameValidator, though suppose this also works

Comment (JeroenDeDauw) [reply tests/Fixtures/NullDomainNameValidator.php]: It's often nice to have such basic test doubles in the package that has the interface. This is now the third place where a SucceedingDomainNameValidator is defined.

image

image

**Comment by JeroenDeDauw** [comment]
No regression test?

**Comment by gbirke** [comment]
The test already existed and the problem was that the *integration* test (which tests with real components) was using a mock instead of the real thing, so it always succeeded.

File: tests/Fixtures/NullDomainNameValidator.php (hunk 1 above)
Comment (gbirke) [reply tests/Fixtures/NullDomainNameValidator.php]: +1

I found your PR https://github.com/wmde/fun-validators/pull/3, will add the `SucceedingDomainNameValidator` there and replace the custom implementations of it wherever I find them.

Where did you get those two other screenshots from?

**Pull request merged**